source/reference/compatibility.txt

```diff
@@ -128,6 +128,32 @@ Write Preference Behavior
 legacy :binary:`mongo` shell. To disable retryable writes, use 
 :option:`--retryWrites=false <mongosh --retryWrites>`.
 
+
+ObjectId Methods and Attributes
+-------------------------------
+
+The ObjectId() methods work differently in ``mongosh`` than in the
+legacy ``mongo`` shell.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Method or Attribute
+     - ``mongo`` Behavior
+     - ``mongosh`` Behavior
+
+   * - ``ObjectId.str``
+     - Hexadecimal string
+     - Undefined
+
+   * - ``ObjectId.valueOf()``
+     - Returns ``ObjectId.str``
+     - Returns ``ObjectId.self``
```

Comment:
I tested this in mongosh and when I ran the .self attribute, I didn't get anything back. Is this expected? Here are the commands I ran:

```
test> x = ObjectId()
ObjectId("6419b14aac4de97027d28de3")

test> x.valueOf()
ObjectId("6419b14aac4de97027d28de3")

test> x.self

(nothing returned)
```